dev/com.ibm.ws.jdbc_fat/fat/src/com/ibm/ws/jdbc/fat/tests/ConfigTest.java

```diff
@@ -1183,18 +1192,21 @@ public void run() {
                     dsfat5.setBeginTranForResultSetScrollingAPIs("false");
                 }
                 updateServerConfig(config, EMPTY_EXPR_LIST);
-                Thread.sleep(100);
+                result = results.poll(POLLING_INTERVAL_MS, TimeUnit.MILLISECONDS);
+                if (result != null && "successful".equals(result)) {
+                    success = true;
+                } else if (result instanceof Throwable) {
+                    throw (Throwable) result;
+                }
             }
         } catch (Throwable x) {
-            System.out.println("Failure during " + method + " with the following config:");
-            System.out.println(config);
+            Log.warning(c, "Failure during " + method + " with the following config:");
+            Log.warning(c, config.toString());
             throw x;
         } finally {
-            Object result = results.poll(10, TimeUnit.SECONDS);
-            if (result == null)
-                throw new Exception("Test did not complete within allotted time");
-            else if (result instanceof Throwable)
-                throw (Throwable) result;
+            if (!success) {
+                throw new Exception("Test testConfigChangeWithActiveConnections did not complete within the allotted time");
+            }
```

Comment:
When `success` is false because the servlet method failed, this finally block obscures the actual failure and replaces it with a more obscure error that will make debugging more difficult.  You should move this code block out of the finally and put at the end of the try block where it is known that no other error has occurred.

Reply:
Good catch!  I have updated that and added the actual timeout value to the timeout exception thrown.